includes/validation/class-amp-validated-url-post-type.php

```diff
@@ -1502,6 +1515,34 @@ public static function print_admin_notice() {
 		}
 	}
 
+	/**
+	 * Render PHP fatal error admin notice.
+	 *
+	 * @param WP_Post $post Post.
+	 */
+	private static function render_php_fatal_error_admin_notice( WP_Post $post ) {
```

Comment:
I'd recommend importing the class `WP_Post` here ( `use WP_Post;` at the top ) because:
- letting PHP guess where the class comes from incurs a performance hit
- using an import makes it clear that the code depends on this class
- using an explicit import makes sure this doesn't break if we refactor the class to be within a namespace (as the relative `WP_Post` would then take on the namespace as well).

Reply:
Does this matter right now since WordPress core isn't using an autoloader? The `WP_Post` class will be a global already.

Also, IntelliJ doesn't like it:

![image](https://user-images.githubusercontent.com/134745/78682932-7c61ad80-78a3-11ea-9dc6-f760a942fbd4.png)

Reply:
Actually, this is also emitting a PHP warning:

> Warning: The use statement with non-compound name 'WP_Post' has no effect in /app/public/content/plugins/amp/includes/validation/class-amp-validated-url-post-type.php on line 9

Reply:
Ah, right, because the `AMP_Validated_URL_Post_Type ` class itself does not use a namespace right now.

Nevermind, then.